nixos/modules/programs/ssh.nix

```diff
@@ -71,22 +149,27 @@ in
         '';
       };
 
-      pubkeyAcceptedKeyTypes = lib.mkOption {
-        type = lib.types.listOf lib.types.str;
-        default = [];
-        example = [ "ssh-ed25519" "ssh-rsa" ];
-        description = ''
-          Specifies the key lib.types that will be used for public key authentication.
-        '';
+      hostSettings = lib.mkOption {
+        type = lib.types.attrsOf settingsNameValueType;
+        description = "Restricts declarations to be only for those hosts that match one of the patterns given.";
+        example = {
+          "myhost.com" = {
+            User = "root";
+            Port = 23;
+          };
+        };
+        default = {};
       };
 
-      hostKeyAlgorithms = lib.mkOption {
-        type = lib.types.listOf lib.types.str;
-        default = [];
-        example = [ "ssh-ed25519" "ssh-rsa" ];
-        description = ''
-          Specifies the host key algorithms that the client wants to use in order of preference.
-        '';
+      matchSettings = lib.mkOption {
+        type = lib.types.attrsOf settingsNameValueType;
```

Comment:
For `Match` blocks, ordering is important, isn't it? That's actually a reason against using attrsOf here.

Reply:
It is, that's why `settingsNameValueType` has a `priority` option ;)

Reply:
You're right, apologies!